iso20022-core/src/main/java/com/prowidesoftware/swift/model/MxSwiftMessage.java

```diff
@@ -329,9 +329,28 @@ private boolean extractMetadata(MxNode n) {
                 updated = true;
             }
         }
+        if (!updated) {
+            // Sender
+            // RITS Reference
+            String requestor = getElement(this.message(), "SwInt:Requestor");
+            String requestorBicString = MxParseUtils.getBICFromDN(requestor);
+            BIC requestorBic = new BIC(requestorBicString);
+            sender = requestorBic.getBic11();
+
+            // Receiver
+            String responder = getElement(this.message(), "SwInt:Responder");
+            String responderBICString = MxParseUtils.getBICFromDN(responder);
+            BIC responderBIC = new BIC(responderBICString);
```

Comment:
esto puede dar null, puede ser que el elemento no exista, y también puede ser que responderBICString sea null porque no se pudo extraer del value. Es decir, se invocaría el constructor de BIC con null. Hay que chequear siempre en cada linea o método de API invocado como trata los null. Aca el sender, si no se puede extraer bien, debe quedar null (not updated). El getBic11 creo que si no tiene datos te tira un default que va a ser misleading

Reply:
Ok

Reply:
Ok